### ISSUE TITLE
Writing to a named pipe is impossible

### ISSUE THREAD

**Issue by thecodewarrior** [exiftool/exiftool#97]
I'm trying to implement file streaming while using `-stay_open` and after some research I found out about named pipes (aka FIFOs), which can be opened like a file but can have their contents streamed from a different process. Using a named pipe as an input works fine, however, trying to output to a named pipe fails since the file already exists. Writing to a named pipe is no different from writing to a file, and a pipe can be detected in Perl using `if (-p $filename)`, so this should be relatively simple to implement. I tried editing the exiftool script to add those checks, but while the script no longer fails, the exiftool library still does.

Before tweaking the script exiftool fails and prints `Error: '_out' already exists - _in`, but after adding the checks it still fails, this time printing `Error: File already exists: _out - _in`.

**Comment by boardhead** [comment]
Does it work if you change line 4124 in ExifTool.pm from this

        return -e $file;

to this

        return(-e $file and not -p $file);

?

**Comment by thecodewarrior** [comment]
Yup, that fixes it!

**Comment by boardhead** [comment]
OK.  I'll add this patch to ExifTool 12.32

- Phil

**Comment by KevinVecchione** [comment]
@boardhead Reading from named pipes works in older versions of `exiftool`, but this change appears to have broken that capability:
```
# exiftool -ver
12.40
# mkfifo foo
# exiftool foo
Error: File not found - foo
```

I am trying to use named pipes with `-stay_open` so that I can stream multiple files as input, so for our use case we can't pipe the image directly to `exiftool`.

**Comment by boardhead** [comment]
Ooops.  OK.  I'll patch the patch. :(

Hopefully this will be fixed in 12.65

**Comment by KevinVecchione** [comment]
Greatly appreciate this and all your hard work.
